### PR TITLE
Allocator: fix experimental makeMultidimensionalArray API

### PR DESCRIPTION
**Pull request by wilzbach**
Unfortunately I was away when #4982 was merged (and no one else has raised their voice either).
However, luckily this hasn't been released :)

Imho we can do provide a better API if

- (1) the type `T` is always the first parameter (after all that's the only template parameter manually provided by the user)
- (2) use variadic arguments and thus let the compiler do the hard work for us

FWIW I really don't like the name and would prefer sth. like `makeNdarray` (that was the name of this method in [`ndslice`](http://dlang.org/phobos/std_experimental_ndslice_slice.html#.makeNdarray).

In short - instead of:

```
size_t[3] dimArray = [2, 3, 6];
auto mArray = Mallocator.instance.makeMultidimensionalArray!(dimArray.length, int)(dimArray);
```

let's better write code like this:

```d
auto mArray = Mallocator.instance.makeMultidimensionalArray!int(2, 3, 6);
```

### CODE REVIEW

File: std/experimental/allocator/package.d [hunk 1]
```diff
@@ -1612,18 +1612,18 @@ lengths = static array containing the size of each dimension
 Returns:
 An N-dimensional array with individual elements of type T.
 */
-auto makeMultidimensionalArray(uint n, T, Allocator)(auto ref Allocator alloc, size_t[n] lengths)
+auto makeMultidimensionalArray(T, Allocator, size_t N)(auto ref Allocator alloc, size_t[N] lengths...)
```
Comment (andralex) [inline std/experimental/allocator/package.d]: Cool. We've had this chataroo in the previous PR - `N` is not exempt from the usual rules: lowercase for values, uppercase for types. So let's make it lowercase. Thanks!

File: std/experimental/allocator/package.d [hunk 2]
```diff
@@ -1612,18 +1612,18 @@ lengths = static array containing the size of each dimension
...
 {
-    static if (n == 1)
+    static if (N == 1)
     {
         return makeArray!T(alloc, lengths[0]);
     }
     else
     {
-        alias E = typeof(makeMultidimensionalArray!(n - 1, T)(alloc, lengths[1..$]));
+        alias E = typeof(makeMultidimensionalArray!(T, Allocator, N - 1)(alloc, lengths[1..$]));
```
Comment (andralex) [inline std/experimental/allocator/package.d]: spaces around `..` :)
